### ISSUE TITLE
Minimal example for SPA implementation of social login

### ISSUE THREAD

**Issue by nimame** [jazzband/dj-rest-auth#147]
First of all many thanks for the great work! I am dedicated to use this package for project I'm working on. However, I have spent a few days now trying to understand the code of allauth and this package in order to build a simple social auth REST API for a SPA app (Angular) and I just fail to understand how a social login (with Google or Azure) works using the `SocialLoginView` (in `/allauth/account/views.py`). This view seems to expose a POST endpoint expecting an `access_token` and/or `code` in its body. But how do I get those in the first place? I would appreciate if someone could post / point me to a minimal example that covers the complete flow from hitting a "Login with Google" button in an SPA app (e.g. Angular, React, Vue or whatever) to receiving the data (e.g. JWT token) necessary to make requests to secured endpoints. I think the docs would greatly benefit from that too. Thanks again!

**Comment by jpaav** [comment]
After struggling to use social logins with my SPA for days, I finally figured out how the flow is supposed to work. Unfortunately I don't have example code, but I can explain what I did to setup Google OAuth2

1. [Create the OAuth2 credentials](https://developers.google.com/identity/protocols/oauth2) and get/make a matching link. [For example, here is what the Google OAuth2 link might look like.](https://developers.google.com/identity/protocols/oauth2/web-server#sample-oauth-2.0-server-response)
2. Put that link in your SPA. When the user clicks it, the "Sign in with Google" page will open.
3. Create a new page in your SPA that will be the callback URL. You will need to specify this URL when you set up the OAuth2 credentials. That page should take the `code` query parameter (or `access_token`) and make a POST request to the `SocialLoginView` or `SocialConnectView` endpoint.
4. Once that post request goes through, you should have usable tokens stored in the `SocialToken` model.

SPA page => Google Sign in page => SPA callback page (`code` as query param) => REST API (`code` as POST data)

The setup for other providers might be a little different, but that's the gist of things.

**Comment by iMerica** [comment]
I agree this would be very helpful. I'm not sure I have the bandwidth to add this, but PRs are welcome!

**Comment by yourfavouriteuncle** [comment]
I'm hitting https://github.com/jezdez/dj-rest-auth/pull/158 and downgrading `djangorestframework` to `3.10.2` did not fix it for me. I'm wondering if there's a good enough workaround because I'm hacking something together right now.
